src/core/lib/channel/call_tracer.h

```diff
@@ -49,11 +49,30 @@ namespace grpc_core {
 // The base class for all tracer implementations.
 class CallTracerAnnotationInterface {
  public:
+  // Enum associated with types of Annotations.
+  enum AnnotationType {
+    kMetadataSizes,
+    kDoNotUse_MustBeLast,
+  };
+
+  // Base class to define a new type of annotation.
+  class Annotation {
+   public:
+    explicit Annotation(AnnotationType type) : type_(type) {}
+    AnnotationType type() const { return type_; }
+    virtual std::string ToString() const = 0;
+    virtual ~Annotation() {}
```

Comment:
we don't need a virtual destructor here... instead make a protected non-virtual destructor.

reasoning: we never want to destroy from the base pointer, we always destroy the most derived type at the callsite, so there's no reason to add bloat (extra vtable entry, extra virtual calls)

Reply:
Done